lib/ansible/modules/remote_management/manageiq/manageiq_provider.py

```diff
@@ -0,0 +1,523 @@
+#!/usr/bin/python
+#
+# (c) 2017, Daniel Korn <korndaniel1@gmail.com>
+#     2017, Yaacov Zamir <yzamir@redhat.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+
+module: manageiq_provider
+
+short_description: Management of provider in ManageIQ.
+extends_documentation_fragment: manageiq
+version_added: '2.4'
+author: Daniel Korn (@dkorn)
+description:
+  - The manageiq_provider module supports adding, updating and deleting provider in ManageIQ.
+
+options:
+  state:
+    description:
+      - absent - provider should not exist, present - provider should be, valid - provider authentication should be valid.
+    required: False
+    choices: ['absent', 'present']
+    default: 'present'
+  name:
+    description:
+      - The provider's name.
+    required: true
+  type:
+    description:
+      - The provider's type.
+    required: true
+    choices: ['Openshift', 'Amazon']
+  zone:
+    description:
+      - The provider's zone name in manageiq.
+    required: false
+    default: 'default'
+  region:
+    description:
+      - The provider's region name in manageiq.
```

Comment:
Both ManageIQ and Amazon have "regions" and "zone".  
IINM, it's zone in manageiq DB and AWS region for Amazon.
=> Please clarify descriptions.
=> Consider `manageiq_zone` and/or `provider_region`?